File: app/src/main/java/br/com/arcom/signpad/dao/AppDataBase.java
package br.com.arcom.signpad.dao;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import br.com.arcom.signpad.dao.converters.DateConverter;
import br.com.arcom.signpad.entities.SigaToken;
import br.com.arcom.signpad.entities.Usuario;
import br.com.arcom.signpad.util.ConstantesUtils;

@Database(
        entities = {
                SigaToken.class,
                Usuario.class
        }
        , version = 1)
@TypeConverters({DateConverter.class})
public abstract class AppDataBase extends RoomDatabase {

    private static AppDataBase dataBase;

    public synchronized static AppDataBase getInstance(Context context) {
        if (dataBase == null) {
            dataBase = Room.databaseBuilder(context.getApplicationContext()
            , AppDataBase.class, ConstantesUtils.DATABASE_NAME)
                    .allowMainThreadQueries()
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return dataBase;
    }

    public abstract SigaTokenDAO sigaTokenDAO();

    public abstract UsuarioDAO usuarioDAO();

}

File: app/src/main/java/br/com/arcom/signpad/database/db/AppDatabase.kt
package br.com.arcom.signpad.database.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import br.com.arcom.signpad.database.converter.AppTypeConverters
import br.com.arcom.signpad.database.dao.LgpdVisitanteDao
import br.com.arcom.signpad.database.entity.LgpdVisitanteEntity
import br.com.arcom.signpad.util.DATABASE_NAME

@Database(
    entities = [
        LgpdVisitanteEntity::class
    ],
    version = 1, exportSchema = false
)
@TypeConverters(AppTypeConverters::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun lgpdVisitanteDao(): LgpdVisitanteDao

    companion object {

        @Volatile
        private var instance: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            return instance ?: synchronized(this) {
                instance ?: buildDatabase(context).also { instance = it }
            }
        }

        private fun buildDatabase(context: Context): AppDatabase {
            return Room.databaseBuilder(context, AppDatabase::class.java, DATABASE_NAME)
                .fallbackToDestructiveMigration()
                .build()
        }
    }
}